### PR TITLE
Add support to load Tiled maps

### PR DESCRIPTION
**Pull request by danielbarnes175**
## Summary
Adds support to load maps created with https://www.mapeditor.org/
Also adds a directory that contains the tiled project

## Technical Changes
- Add various mapping classes in `/Engine/Map/` that mirror the Tiled JSON structure, so that we can directly load into it
- Combine Program.cs into Game1.cs and renamed Main.cs to make it cleaner
- Add example loading to Main.cs
- Add Tiled map directory, outside of the main game directory so we don't export these files when publishing
- Add Newtonsoft.Json as a dependency to make loading JSON easier

## Screenshots
image

### CODE REVIEW

File: GateAdventures/Engine/Map/TiledMapLoader.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+			LoadSingleMap(content, graphics, "assets/maps/exampleMap.json")
+		};
+
+		return loadedMaps;
+	}
+
+	public static TileMapRenderer LoadSingleMap(ContentManager content, GraphicsDevice graphics, string mapFilePath)
+	{
+		// Load JSON into TileMap object
+		string filePath = Path.Combine(content.RootDirectory, mapFilePath);
```
Comment (maxdevos49) [inline GateAdventures/Engine/Map/TiledMapLoader.cs]: When I run `dotnet run --project GateAdventure` it fails to find the JSON file.

The resulting path seems to be: `<path-to-where-command-is-run>` + `<content.RootDirectory>` + `<mapFilePath>`.

If I run `dotnet run` inside the GateAdventures folder it works. I think it should work in both situations and not require being inside the Project folder. Desired path ending up being: `<ExecutableFolderPath>` + `<content.RootDirectory>` + `<mapFilePath>`.

File: GateAdventures/Game1.cs [hunk 2]
```diff
@@ -1,51 +0,0 @@
-﻿using Microsoft.Xna.Framework;
```
Comment (maxdevos49) [inline GateAdventures/Game1.cs:1]: I agree I am not a fan of this class being named Game1 but I did like the Program.cs file. What if the `Game1` class was renamed `GateAdventures` and Program.cs was preserved?

File: GateAdventures/Engine/Map/Layers.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
```
Comment (maxdevos49) [inline GateAdventures/Engine/Map/Layers.cs:1]: Mostly a nit but I think we should drop `Engine` in this file path

**Review by maxdevos49 — DISMISSED** [review]
@danielbarnes175 can you add some brief instructions on how to use the tile map editor to the readme

File: GateAdventures/Engine/Map/Layers.cs (hunk 3 above)
Comment (maxdevos49) [reply GateAdventures/Engine/Map/Layers.cs:1]: I retract this

Comment (danielbarnes175) [reply GateAdventures/Engine/Map/Layers.cs:1]: Dang it!!

**Comment by danielbarnes175** [comment]
@maxdevos49 Comments addressed,
- Added program.cs back in, renamed Main.cs to GateAdventures.cs
- Adjusted the path of the file to use `Path.Combine(AppContext.BaseDirectory, content.RootDirectory, mapFilePath);`
- Got rid of the `Engine` folder and added it back 🙊 
- Added a new `README.md` in the Tiled folder since it made more sense to me to have it separate from the main one. There were also some additional notes I needed to add about how to actually import the map

File: GateAdventures/Engine/Map/Layers.cs (hunk 3 above)
Comment (maxdevos49) [reply GateAdventures/Engine/Map/Layers.cs:1]: Sorry D:

**Review by maxdevos49 — APPROVED** [review]

**Pull request merged**